src/core/types/keys/private_key.rs

```diff
@@ -44,13 +44,42 @@ impl PrivateKey {
     /// in the `PublicKey` tagged union.
     pub fn public_key(&self) -> PublicKey {
         match self {
-            PrivateKey::Ed25519(key) => PublicKey::Ed25519 {
-                value: key.public_key(),
+            PrivateKey::Ed25519(key) => PublicKey::Ed25519(key.public_key()),
+            PrivateKey::Secp256k1(key) => {
+                PublicKey::Secp256k1(key.public_key())
+            }
+        }
+    }
+
+    pub fn sign(&self, msg_hash: &Hash) -> Signature {
```

Comment:
same method that existed on `Secp256k1PrivateKey` and `Ed25519PrivateKey` but on the general type `PrivateKey`.